src/main/ruby/core/rbconfig.rb

```diff
@@ -171,6 +171,7 @@ module RbConfig
                     '-Wno-int-conversion',                    # MRI has VALUE defined as long while we have it as void*
                     '-Wno-int-to-pointer-cast',               # Same as above
                     '-Wno-unused-value',                      # RB_GC_GUARD leaves
+                    '-Wno-incompatible-pointer-types',        # We define VALUE as void* rather than uint32_t
```

Comment:
@eregon I'm proposing adding this flag back that was removed in eddf0f5f2d7b1be7dda8e062e6656917c18d8e9e to get this version of byebug to compile.

Reply:
OK, but then please explain why it is needed in the comment.

> I'm seeing this for byebug `8.2.1`
```
threads.c:119:38: error: incompatible pointer types passing 'VALUE *' (aka 'void **') to parameter of type 'st_data_t *' (aka 'unsigned long *')
      [-Werror,-Wincompatible-pointer-types]
  if (!st_lookup(t_tbl->tbl, thread, context) || !*context)
                                     ^~~~~~~
.../truffleruby/lib/cext/ruby/st.h:115:49: note: passing argument to parameter here
int st_lookup(st_table *, st_data_t, st_data_t *);
```
> the extconf.rb for that version is `-Wall -Werror`.

Reply:
So this is an example of a flag we want for gem C exts but probably not internal C exts (zlib had a bug due to this flag ignoring an actual problem).

Reply:
Updated the comment.